server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -954,11 +964,14 @@ public static String findV2Template(Metadata metadata, String indexName, boolean
             .map(ComponentTemplate::template)
             .map(Template::mappings)
             .filter(Objects::nonNull)
-            .collect(Collectors.toList());
+            .collect(Collectors.toCollection(LinkedList::new));
         // Add the actual index template's mappings, since it takes the highest precedence
         Optional.ofNullable(template.template())
             .map(Template::mappings)
             .ifPresent(mappings::add);
+        // add a default mapping for the `@timestamp` field, at the lowest precedence, to make bootstrapping data streams more
+        // straightforward
+        mappings.add(0, new CompressedXContent(wrapMappingsIfNecessary(DEFAULT_TIMESTAMP_MAPPING, xContentRegistry)));
```

Comment:
Shouldn't this be inside an `if` to only add this if the template has a data stream component defined?

Reply:
I was under the impression we always wanted to add it, however I think you're right. Thanks, Lee